rviz_common/src/rviz_common/interaction/selection_manager.cpp

```diff
@@ -267,14 +267,17 @@ void SelectionManager::unpackColors(const Ogre::PixelBox & box)
   pixel_buffer_.clear();
   pixel_buffer_.reserve(w * h);
 
-  for (uint32_t y = 0; y < h; ++y) {
-    for (uint32_t x = 0; x < w; ++x) {
-      uint32_t pos = (x + y * w) * 4;
-
-      uint32_t pix_val = *reinterpret_cast<uint32_t *>(static_cast<uint8_t *>(box.data) + pos);
-      uint32_t handle = colorToHandle(box.format, pix_val);
-
-      pixel_buffer_.push_back(handle);
+  size_t size = Ogre::PixelUtil::getMemorySize(1, 1, 1, box.format);
+
+  for (unsigned int y = 0; y < h; y++) {
+    for (unsigned int x = 0; x < w; x++) {
```

Comment:
```suggestion
  for (uint32_t y = 0; y < h; y++) {
    for (uint32_t x = 0; x < w; x++) {
```

(as far as I can tell, these are always uint32_t in https://github.com/OGRECave/ogre/blob/master/OgreMain/include/OgreCommon.h#L789-L791)

(it would also be awesome to use `uint32_t` for `w` and `h` above)